ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py

```diff
@@ -1,29 +1,129 @@
 # Copyright (c) 2023-2024 Arista Networks, Inc.
 # Use of this source code is governed by the Apache License 2.0
 # that can be found in the LICENSE file.
+from copy import deepcopy
 from functools import lru_cache
-from pickle import load
+from hashlib import sha1
+from pickle import HIGHEST_PROTOCOL
+from pickle import dump as pickle_dump
+from pickle import load as pickle_load
 
 from yaml import safe_load
 
-from ansible_collections.arista.avd.plugins.plugin_utils.schema.default_schemas import DEFAULT_PICKLED_SCHEMAS, DEFAULT_SCHEMAS
+from .avdschemaresolver import AvdSchemaResolver
+from .default_schemas import DEFAULT_PICKLED_SCHEMAS, DEFAULT_SCHEMAS
 
 
 @lru_cache
-def create_store(load_from_yaml=False):
+def create_store(load_from_yaml=False, force_rebuild=False):
     store = {}
 
     # Load from YAML if set. This is used by the tool that creates the pickle.
     if load_from_yaml:
-        for id, schema_file in DEFAULT_SCHEMAS.items():
-            with open(schema_file, "r", encoding="UTF-8") as file:
-                schema_file_data = safe_load(file.read())
-                store[id] = schema_file_data
+        return _create_store_from_yaml()
 
-    # Load from Pickle
-    else:
-        for id, schema_file in DEFAULT_PICKLED_SCHEMAS.items():
-            with open(schema_file, "rb") as file:
-                store[id] = load(file)
+    # Recompile schemas if needed and store in pickles for next time.
+    elif force_rebuild or _should_recompile_schemas():
+        return _compile_schemas()
+
+    # Load from Pickle.
+    for id, schema_file in DEFAULT_PICKLED_SCHEMAS.items():
+        with open(schema_file, "rb") as file:
+            store[id] = pickle_load(file)
 
     return store
+
+
+def _should_recompile_schemas() -> bool:
+    """
+    Returns true if pickled schemas should be recompiled
+    """
+    # Check if any pickled schema is missing
+    for id, pickle_file in DEFAULT_PICKLED_SCHEMAS.items():
+        if not pickle_file.exists():
+            return True
+    # Check if any hash file is missing
```

Comment:
Using values instead of items as not using id + using any
```suggestion
if any(not pickle_file.exists() for pickle_file in DEFAULT_PICKLED_SCHEMAS.values()):
  return True
```

Reply:
reading the rest, actually rather than doing three loops, why not just do the three checks in a single loop?

Reply:
I am looping over different files. And to avoid doing 3 checks on each I wanted to exit quickly for the most common case.

Reply:
the outer loop is for the same files

Reply:
yes but since I replace in one, the check is not for the same file.

Reply:
I wanted to avoid running the hash operation unless strictly needed, but I have combined the last two to make you happy :)